src/libcore/slice/mod.rs

```diff
@@ -5419,6 +5453,151 @@ unsafe impl<'a, T> TrustedRandomAccess for ChunksExactMut<'a, T> {
     }
 }
 
+/// An iterator over a slice in (non-overlapping) chunks (`N` elements at a
+/// time), starting at the beginning of the slice.
+///
+/// When the slice len is not evenly divided by the chunk size, the last
+/// up to `chunk_size-1` elements will be omitted but can be retrieved from
+/// the [`remainder`] function from the iterator.
+///
+/// This struct is created by the [`array_chunks`] method on [slices].
+///
+/// [`array_chunks`]: ../../std/primitive.slice.html#method.array_chunks
+/// [`remainder`]: ../../std/slice/struct.ArrayChunks.html#method.remainder
+/// [slices]: ../../std/primitive.slice.html
+#[derive(Debug)]
+#[unstable(feature = "array_chunks", issue = "none")]
+pub struct ArrayChunks<'a, T: 'a, const N: usize> {
+    v: &'a [T],
```

Comment:
Another possibility is to cast `&[T]` to `&[[T; N]]` up front, then store `Iter<'a, [T; N]>` here. That will at least simplify the `Iterator` implementation, making it totally safe.

Reply:
That's a good idea! Thanks :heart:

Reply:
Can't derive `Debug` in this case. Will add this as a `FIXME` as it is otherwise a lot cleaner and update it once #74060 lands